src/derivative_wrappers.jl

```diff
@@ -431,7 +431,7 @@ function _vecjacobian!(dλ, y, λ, p, t, S::TS, isautojacvec::ReverseDiffVJP, dg
 
     if prob isa Union{SteadyStateProblem, NonlinearProblem} ||
        (eltype(λ) <: eltype(prob.u0) && typeof(t) <: eltype(prob.u0) &&
-        compile_tape(sensealg.autojacvec))
+        compile_tape(sensealg.autojacvec) && !recompile_tape)
```

Comment:
why here?

Reply:
this is the place where we check if the compiled tape should be used. So I added the additional boolean to avoid using the old tape and build a new one when needed for the callback handling.